### ISSUE TITLE
Codewise creation of Pins requires multiple map clicks before event fires !

### ISSUE THREAD

**Issue by sailman342** [Mapsui/Mapsui#791]
Creating (say 10) pins by code at View initailisation (or onappearing event) will require the user to click (10) times before the mapClicked event fires !

This is true with 2.0 beta 32, with build 969 first click will crash (hang)

See sample demo in my Dropbox.

Impossible to setup the callout, uncommenting code of the exemple will provoque hang as well !

```
`
        protected override void OnAppearing()
        {
            for (int index = 0; index < 10; index++)
            {
                Pin tempPin = new Pin(MyMapView)
                {
                    Label = "Native OSM Pin  " + index.ToString(),
                    Address = "Quelque part !",
                    Position = new Position(initLoc.Latitude + index * 0.0003, initLoc.Longitude + index * 0.003),
                    Type = PinType.Pin,
                    Color = Color.Red,
                    Transparency = 0.5f,
                    Scale = 1.0f
                };

                /* thid code crashes
                
                // crashes with or without setting the type
                //tempPin.Callout.Type = CalloutType.Detail;
                tempPin.CalloutAnchor = new Point(0, tempPin.Height * tempPin.Scale);
                tempPin.Callout.RectRadius = 5.0f;
                tempPin.Callout.ArrowHeight = 10.0f;
                tempPin.Callout.ArrowWidth = 10.0f;
                tempPin.Callout.ArrowAlignment = (ArrowAlignment)2;
                tempPin.Callout.ArrowPosition = 0.5f;
                tempPin.Callout.SubtitleLabel.LineBreakMode = LineBreakMode.NoWrap;

                */

                thePins.Push(tempPin);
                MyMapView.Pins.Add(tempPin);
            }

            MyMapView.Refresh();

        }
`
```

https://www.dropbox.com/s/fkypr4mvmy5m25f/MultipleNOPs.zip?dl=0

**Comment by pauldendulk** [comment]
I was aware of this but was not able to fix this last Friday.

**Comment by pauldendulk** [comment]
It is indeed related to the callout refactoring.

**Comment by pauldendulk** [comment]
I was aware of the hang caused by 2.0.0-build.969. This is related to the callout.

**Comment by sailman342** [comment]
Yes but if you load a map with ten's of pins it is a nightmare before you can use it :-)

I Don't know if the issues are related !

But anyway thanks for your efforts, the Dropbox sample reproduces easyly the problem !

**Comment by sailman342** [comment]
Well, I am a bit disoriented !

I downloaded the Mapsui sources from Github, inserted my app as a new project, removed the links to the nuget packages and Added references to the Mapsui project instead ! 

The problem is not hsowing anymore ? 

I went up to package a distribution of my project and installed it on my Windows 10 PC and it works fine ! Are the Nuget 2.0 Beta 32 sources different from the GitHub ones ?

Kindest , Alain

**Comment by charlenni** [comment]
Sure, there are some PRs in the last days merged into the master.

**Comment by pauldendulk** [comment]
The 2.0.0-build.xxx packages are directly built from the latest commit to master.
https://mapsui.com/documentation/NuGet-of-latest-build.html

The 2.0.0-beta.xx packages are explicitly released from time to time. They correspond to the taggs
https://github.com/Mapsui/Mapsui/tags
And there is always a commit which updates the version number.
https://github.com/Mapsui/Mapsui/commit/fc0e0a907080fd0138e366ba6d6d6ed4d06bc7cd

Could this explain what you observe?

**Comment by sailman342** [comment]
I used the master from here https://github.com/Mapsui/Mapsui !

Is there a way to get the commit number from downloaded sources ?

I would not like to bother you too much guys on the real job :-) 

What is your recommandation to use for helping you a maximum by reporting you real problems … ?

Should I use the latest build or the latest beta ?

Keep going :-)

**Comment by pauldendulk** [comment]
Usually both can be useful but right now there are known issues with the callout and changes are in progress.

**Comment by pauldendulk** [comment]
We made some changes to the callout mechanism and assume this is fixed in 2.0.0-beta.34. Please let us know when it is not.
